score-lang-content-verifier/src/main/java/org/openscore/lang/tools/verifier/VerifierHelper.java

```diff
@@ -0,0 +1,134 @@
+ */
+
+/**
+ * Verifies all files with extensions: .sl, .sl.yaml or .sl.yml in a given directory are valid
+ */
+public class VerifierHelper {
+
+    @Autowired
+    private SlangCompiler slangCompiler;
+
+    @Autowired
+    private ScoreCompiler scoreCompiler;
+
+    private final static Logger log = Logger.getLogger(SlangFilesVerifier.class);
+
+    private String[] SLANG_FILE_EXTENSIONS = {"sl", "sl.yaml", "sl.yml"};
+
+    private Map<String, Executable> slangModels = new HashMap<>();
+
+    private Map<String, CompilationArtifact> compiledArtifacts = new HashMap<>();
+
+    /**
+     * Transform all Slang files in given directory to Slang models, and store them
+     * @param directoryPath given directory containing all Slang files
+     */
+    public void verifyAllSlangFilesInDirAreValid(String directoryPath){
+        transformSlangFilesInDirToModels(directoryPath);
+        compileAllSlangModels();
+    }
+
+    private void transformSlangFilesInDirToModels(String directoryPath) {
+        Validate.notNull(directoryPath, "Directory path path can not be null");
+        Collection<File> slangFiles = FileUtils.listFiles(new File(directoryPath), SLANG_FILE_EXTENSIONS, true);
+        log.info("Start compiling all slang files under: " + directoryPath);
+        log.info(slangFiles.size() + " .sl files were found");
+        for(File slangFile: slangFiles){
+            Validate.isTrue(slangFile.isFile(), "filePath must lead to a file");
+            Executable sourceModel;
+            try {
+                sourceModel = slangCompiler.preCompile(fromFile(slangFile));
+            } catch (Exception e) {
+                log.error("Failed creating Slang ,models for directory: " + directoryPath + ". Exception is : " + e.getMessage());
+                throw e;
+            }
+            verifyStaticCode(slangFile, sourceModel);
+            slangModels.put(getUniqueName(sourceModel), sourceModel);
+        }
+        if(slangFiles.size() != slangModels.size()){
+            throw new RuntimeException("We found: " + slangFiles.size() + " .sl files in path: " + directoryPath + ". We managed to create slang models for only: " + slangModels.size());
+        }
+    }
+
+    private void compileAllSlangModels()  {
+        Collection<Executable> models = slangModels.values();
+        for(Executable slangModel : models) {
+            try {
+                Set<Executable> dependenciesModels = getModelDependenciesRecursively(slangModel);
+                CompilationArtifact compiledSource = compiledArtifacts.get(slangModel.getName());
+                if (compiledSource == null) {
+                    compiledSource = scoreCompiler.compile(slangModel, dependenciesModels);
+                    log.info("Compiled: " + slangModel.getName() + " successfully");
+                    compiledArtifacts.put(slangModel.getName(), compiledSource);
+                } else {
+                    log.error("Failed to compile: " + slangModel.getName());
```

Comment:
why? 
doesn't it just means that is was already compiled. (and not possible since it is a Set)